drivers: can: tcan4x5x: Build fails/emits warnings when optional reset-gpios property is not supplied in device tree

<!--
**Notes**
Github Discussions (https://github.com/zephyrproject-rtos/zephyr/discussions)
are available to first verify that the issue is a genuine Zephyr bug and not a
consequence of Zephyr services misuse.

This issue list is only for bugs in the main Zephyr code base
(https://github.com/zephyrproject-rtos/zephyr/). If the bug is for a project
fork (such as NCS) specific feature, please open an issue in the fork project
instead.
-->

**Describe the bug**
<!--
A clear and concise description of what the bug is.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - What have you tried to diagnose or workaround this issue?
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 - ...
-->
Build fails when building with `ti,tcan4x5x`-compatible node that doesn't have `reset-gpios` property supplied because of unused local variables in `tcan4x5x_reset(const struct device *dev)` function. Usually compiler emits a warning here, but I am building with Sysbuild and `SB_CONFIG_COMPILER_WARNINGS_AS_ERRORS=y`, so this becomes an error.

**To Reproduce**
<!--
Steps to reproduce the behavior:
1. mkdir build; cd build
2. cmake -DBOARD=board\_xyz
3. make
4. See error
-->

1. Add this node as a child node to SPI controller:
```
    can0: tcan4550@0 {
        status = "okay";
        compatible = "ti,tcan4x5x";
        reg = <0>;
        spi-max-frequency = <DT_FREQ_M(16)>;
        int-gpios = <&gpio0 1 GPIO_ACTIVE_LOW>;
        clock-frequency = <DT_FREQ_M(40)>;
        bosch,mram-cfg = <0x0 15 15 5 5 0 10 10>;

        can-transceiver {
            status = "okay";
            compatible = "can-transceiver";
            #phy-cells = <0>;
            max-bitrate = <DT_FREQ_M(8)>;
        };
    };
```

2. Add `CONFIG_CAN=y` to your project's `prj.conf` file
3. Add `SB_CONFIG_COMPILER_WARNINGS_AS_ERRORS=y` to your project's `sysbuild.conf` file.
4. Try to build the project with Sysbuild

**Expected behavior**
<!--
A clear and concise description of what you expected to happen.
-->
The project should be buildable without having to supply `reset-gpios` property to `ti,tcan4x5x`-compatible node, since device reset can be done by an SPI command.

**Impact**
<!--
What impact does this issue have on your progress (e.g., annoyance, showstopper)
-->
Annoyance, since you can provide a "dummy" GPIO in device tree.

**Logs and console output**
<!--
If applicable, add console logs or other types of debug information
e.g Wireshark capture or Logic analyzer capture (upload in zip archive).
copy-and-paste text and put a code fence (\`\`\`) before and after, to help
explain the issue. (if unable to obtain text log, add a screenshot)
-->
```
/home/tmilkovic/zephyrproject/zephyr/drivers/can/can_tcan4x5x.c: In function 'tcan4x5x_reset':
/home/tmilkovic/zephyrproject/zephyr/drivers/can/can_tcan4x5x.c:512:39: error: unused variable 'tcan_config' [-Werror=unused-variable]
  512 |         const struct tcan4x5x_config *tcan_config = mcan_config->custom;
      |                                       ^~~~~~~~~~~
cc1: all warnings being treated as errors
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**

 - OS: Linux (Manjaro)
 - Toolchain: Zephyr SDK 0.17.0
 - Commit SHA or Version used: Current main (9a236f82c185524edf901d576c46f874ca5b78d2)

**Additional context**
<!--
Add any other context that could be relevant to your issue, such as pin setting,
target configuration, ...
-->
Used custom board with `ti,tcan4x5x`-compatible node, but issue can be recreated by applying `tcan4550evm` shield to your board with `reset-gpios` property deleted.

--- a/drivers/can/can_tcan4x5x.c
+++ b/drivers/can/can_tcan4x5x.c
@@ -508,8 +508,10 @@ static int tcan4x5x_wake(const struct device *dev)
 
 static int tcan4x5x_reset(const struct device *dev)
 {
+#if TCAN4X5X_RST_GPIO_SUPPORT
 	const struct can_mcan_config *mcan_config = dev->config;
 	const struct tcan4x5x_config *tcan_config = mcan_config->custom;
+#endif /* TCAN4X5X_RST_GPIO_SUPPORT */
 	int err;
 
 	err = tcan4x5x_wake(dev);

